presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/ValidateDependenciesChecker.java

```diff
@@ -598,6 +599,25 @@ public Void visitApply(ApplyNode node, Set<Symbol> boundSymbols)
             return null;
         }
 
+        @Override
+        public Void visitLateralJoin(LateralJoinNode node, Set<Symbol> boundSymbols)
+        {
+            Set<Symbol> subqueryCorrelation = ImmutableSet.<Symbol>builder()
+                    .addAll(boundSymbols)
+                    .addAll(node.getCorrelation())
+                    .build();
+
+            node.getInput().accept(this, boundSymbols); // visit child
+            node.getSubquery().accept(this, subqueryCorrelation); // visit child
+
+            checkDependencies(node.getInput().getOutputSymbols(), node.getCorrelation(), "LATERAL input must provide all the necessary correlation symbols for subquery");
+            checkDependencies(DependencyExtractor.extractUnique(node.getSubquery()), node.getCorrelation(), "not all APPLY correlation symbols are used in subquery");
```

Comment:
remove `APPLY` from message: `not all APPLY correlation symbols are used in subquery`.